services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,55 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
+  if(thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdPASS){
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  //define new event type to send to thermal manager queue 
+  thermal_mgr_event_t sendEvent;
+  sendEvent.type = THERMAL_MGR_EVENT_INTERRUPT_TRIGGERED;
+  thermalMgrSendEvent(&sendEvent);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t recieved;
+    float temp;
+    error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+
+    //wait until item is available in queue
+    if(xQueueReceive(thermalMgrQueueHandle, &recieved, portMAX_DELAY) == pdPASS){
+      if(errCode == ERR_CODE_SUCCESS){
+        if(recieved.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+          addTemperatureTelemetry(temp);
+        }
+        else if(recieved.type == THERMAL_MGR_EVENT_INTERRUPT_TRIGGERED){
+          if(temp > LM75BD_DEFAULT_HYST_THRESH){
+            overTemperatureDetected();
+          }
+          else{
+            safeOperatingConditions();
+          }
+        }
+      }
+      else{
+        LOG_ERROR_CODE(errCode);
+      }
+    }
+    else{
+      LOG_ERROR_CODE(ERR_CODE_QUEUE_FULL);
```

Comment:
A few problems here:
- Since you use portMAX_DELAY as the block time, this task will be suspended indefinitely until it receives an event. I.e. if xQueueReceive returns, you can be certain that an event was received. See https://www.freertos.org/a00118.html (we have INCLUDE_vTaskSuspend set to 1).
- If xQueueReceive used a finite block time, then it'd only fail if nothing was in the queue. I.e. xQueueReceive would never fail if the queue was full. This is a misleading error code to log.